### PR TITLE
Decompile Axelord Shield func_ptr_80170024

### PR DESCRIPTION
**Pull request by bismurphy**
Next one! Lots of collisions here which is interesting.

### CODE REVIEW

File: src/weapon/w_011.c [hunk 1]
```diff
@@ -155,8 +158,218 @@ int GetWeaponId(void) { return 11; }
...
+            yOffset = 8;
+            xVar = self->posX.i.hi + xOffset;
+            yVar = self->posY.i.hi + yOffset;
+            g_api.CheckCollision(xVar, yVar, &collider, 0);
+            if ((collider.effects & EFFECT_SOLID) &&
+                !(collider.effects & EFFECT_UNK_8000)) {
+                goto make_other_entity;
+            }
+        }
+        if ((((collider.effects & (EFFECT_UNK_8000 | EFFECT_UNK_4000)) ==
```
Comment (Xeeynamo) [inline src/weapon/w_011.c:304]:
```suggestion
        if ((((collider.effects & (EFFECT_UNK_8000 | EFFECT_UNK_4000)) ==
```
I wonder if extracting `(EFFECT_UNK_8000 | EFFECT_UNK_4000)` into a `const` or `#define` would make that if statement less of a nightmare to read

File: src/weapon/w_011.c [hunk 2]
```diff
@@ -155,8 +158,218 @@ int GetWeaponId(void) { return 11; }
...
+        } else {
+            goto make_other_entity;
+        }
+        break;
+    case 3:
+        if (self->animFrameDuration >= 0) {
+            break;
+        }
+        DestroyEntity(self);
+        return;
```
Comment (Xeeynamo) [inline src/weapon/w_011.c:353]:
```suggestion
        if (self->animFrameDuration < 0) {
            DestroyEntity(self);
            return;
        }
            break;
```
flipping this for consistency. I did not test if this matches

**Review by Xeeynamo — APPROVED** [review]

**Review by sozud — APPROVED** [review]

File: src/weapon/w_011.c (hunk 2 above)
Comment (bismurphy) [reply src/weapon/w_011.c:353]: Ah yes good point of course, will swap

File: src/weapon/w_011.c (hunk 1 above)
Comment (bismurphy) [reply src/weapon/w_011.c:304]: Maybe I'll just make an UNK_C000 and then we can name it properly once we know what each of the individual 8000 and 4000 are

**Pull request merged**